File: IAIP.Testx/ClassTests/ReverseComparerTests.cs
using System;
using System.Collections.Generic;
using Iaip;
using Xunit;

namespace IAIP.Testx.ClassesTests
{
    public class ReverseComparerTests
    {
        private SortedList<int, string> mySortedList;
        private SortedList<int, string> reverseSortedList;

        public ReverseComparerTests()
        {
            Dictionary<int, string> dict = new Dictionary<int,string> 
            {
                {1, "abc"},
                {2, "def"},
                {3, "ghi"}
            };

            mySortedList = new SortedList<int, string>(dict);

            reverseSortedList = new SortedList<int, string>(dict, new ReverseComparer<int>(Comparer<int>.Default));
        }

        [Fact]
        public void CompareEachElement()
        {
            int count = mySortedList.Count;

            Assert.Equal(count, reverseSortedList.Count);

            for (int i = 0; i < mySortedList.Count; i++)
            {
                Assert.Equal(mySortedList.Values[i], reverseSortedList.Values[count - 1 - i]);
            }
        }
    }
}

File: IAIP.Test/ClassTests/ReverseComparerTests.vb
Imports Xunit

Public Class ReverseComparerTests

    <Fact>
    Public Sub CompareEachElement()
        Dim dict As New Dictionary(Of Integer, String) From {
            {1, "abc"},
            {2, "def"},
            {3, "ghi"}
        }

        Dim mySortedList As New SortedList(Of Integer, String)(dict)
        Dim reverseSortedList As New SortedList(Of Integer, String)(dict, New ReverseComparer(Of Integer)(Comparer(Of Integer).Default))

        Dim count As Integer = mySortedList.Count

        Assert.Equal(count, reverseSortedList.Count)

        For i As Integer = 0 To mySortedList.Count - 1
            Assert.Equal(mySortedList.Values(i), reverseSortedList.Values(count - 1 - i))
        Next
    End Sub

End Class
